### PR TITLE
Set Artifactory version 5.8

### PR DESCRIPTION
**Pull request by eldada**
*****************************************************************************************

Thank you for contributing to kubernetes/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the 
history. This will make it easier to identify new changes. The PR will be squashed 
anyways when it is merged. Thanks.

*****************************************************************************************

### CODE REVIEW

File: stable/artifactory/values.yaml
Comment (mattfarina) [inline stable/artifactory/values.yaml]: A version should be listed to pin the version for reproducible deployments

**Review by mattfarina — CHANGES_REQUESTED** [review]

File: stable/artifactory/values.yaml
Comment (eldada) [reply stable/artifactory/values.yaml]: I was actually thinking that having the `Chart.yaml's` `appVersion` as a single source of version covers it.
I have a template in the `_helpers.tpl` called "artifactory.artifactory.version", which is used in the artifactory-deployment.yaml:
`image: '{{ .Values.artifactory.image.repository }}:{{ template "artifactory.artifactory.version" . }}'`
This way, I don't need to change the version in multiple files.
Did I miss something?

Comment (mattfarina) [reply stable/artifactory/values.yaml]: I didn't see that. I like it.

Prior to merging this I would like to have a quick chat with the other maintainers.

/hold
/assign

**Comment by eldada** [comment]
@mattfarina - any estimate on this? Any open issues I should address?
We have a few issues pending the completion of this PR.

**Comment by mattfarina** [comment]
@eldada The meeting where we discuss these sorts of things is on Tuesday of this week. The holidays caused us to go a little longer than normal without meeting. I should have some direction by Wednesday.

File: stable/artifactory/values.yaml
Comment (mattfarina) [reply stable/artifactory/values.yaml]: /hold cancel

@eldada we had a conversation about this in the charts chat call today. Several maintainers weighed in. Thank you for being patient with us while we talked this out.

What you've done with the helpers is interesting and on more complicated charts, with multiple images using the same version, could make updates easier. This does make updating the chart easier for maintainers.

But, for those consuming the chart it makes it a little more difficult. What version being used isn't in the README. How you can set it yourself isn't intuitive. I've already witnessed a few people trip over setting their own tag version.

Because we want to prioritize chart consumers and their experience we would prefer to have explicit versions in the values.yaml file instead of pulling a default from the Chart.yaml file. We understand it's a little more work for chart maintainers and there's some duplication. But, it appears to be more intuitive for consumers.

Comment (eldada) [reply stable/artifactory/values.yaml]: @mattfarina - I understand. I will revert this to the old format.
I think there should be an effort towards a best practice of a single source for the app version, that is used across all chart. I'll let you and the team play with this idea.
Maybe it's just an issue of better documentation (?)

**Comment by eldada** [comment]
@mattfarina - ready for your final review.

File: stable/artifactory/README.md
Comment (mattfarina) [inline stable/artifactory/README.md]: In the docs we'd like to see the version (5.8.2 in this case)

File: stable/artifactory/values.yaml
Comment (mattfarina) [inline stable/artifactory/values.yaml:101]: What is someone wanted to override the nginx or artifactory images? They would have change any version tags as a group rather than one or the other.

And, if someone reads the values.yaml file as a consumer of the chart they may be confused on setting the version.

Could you please revert where the versions are combined?

**Review by mattfarina — COMMENTED** [review]
Just a couple more nits

**Review by mattfarina — APPROVED** [review]

**Comment by mattfarina** [comment]
@eldada I think the github outage had an effect on the CI results reporting. The cla and circleci ones are waiting for a response. Is there any way you could push a new commit that does basically nothing to kick these off. I'm sorry for the inconvenience. Seems this communication between CI tools and GitHub isn't fault tolerant enough.

**Comment by eldada** [comment]
@mattfarina - What is the "tide" check? What does it mean "Not in tide pool"?

**Pull request merged**
